app/services/registration_service.rb

```diff
@@ -1,7 +1,10 @@
 class RegistrationService
   def from_invitation(email, invitation_token, user_data = {})
-    Invitation.find_by(email: email, token: invitation_token)
-    User.create({ email: email }.merge(user_data))
+    invitation = Invitation.find_by(email: email, token: invitation_token)
+    user = User.where(email: email).first
```

Comment:
that will raise exception if not found.. :cry:
On Jun 13, 2015 8:52 PM, "GpoNsU" <notifications@github.com> wrote:

> In app/services/registration_service.rb
> <https://github.com/TheTreeHouse/turniquette-api/pull/51#discussion_r32372815>
> :
>
> > @@ -1,7 +1,10 @@
> >  class RegistrationService
> >    def from_invitation(email, invitation_token, user_data = {})
> > -    Invitation.find_by(email: email, token: invitation_token)
> > -    User.create({ email: email }.merge(user_data))
> > +    invitation = Invitation.find_by(email: email, token: invitation_token)
> > +    user = User.where(email: email).first
>
> SUG:
>
> user = User.find_by(email: email)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TheTreeHouse/turniquette-api/pull/51/files#r32372815>.
>